src/main/java/ch/njol/util/Math2.java

```diff
@@ -163,4 +161,16 @@ public static long multiplyClamped(long x, long y) {
 		return result;
 	}
 
+	public static @Nullable Long factorial(long x) {
```

Comment:
This is probably quite dangerous because it's very likely to overflow (for, like, most numbers) and there's no detection for that or warning to the user and their result will just be incorrect.